src/interactive-auth.js

```diff
@@ -201,8 +339,30 @@ InteractiveAuth.prototype = {
      */
     _chooseFlow: function() {
         const flows = this._data.flows || [];
-        // always use the first flow for now
-        return flows[0];
+
+        // we've been given an email or we've already done an email part
+        const haveEmail = Boolean(this._inputs.emailAddress) || Boolean(this._emailSid);
+        const haveMsisdn = (
+            Boolean(this._inputs.phoneCountry) &&
+            Boolean(this._inputs.phoneNumber)
+        );
+
+        for (const flow of flows) {
+            let flowHasEmail = false;
+            let flowHasMsisdn = false;
+            for (const stage of flow.stages) {
+                if (stage === EMAIL_STAGE_TYPE) {
+                    flowHasEmail = true;
+                } else if (stage == MSISDN_STAGE_TYPE) {
+                    flowHasMsisdn = true;
+                }
+            }
+
+            if (flowHasEmail == haveEmail && flowHasMsisdn == haveMsisdn) {
```

Comment:
Well, the code previously worked as you suggest, but I realised that, at least in the case of registration, if the user provided, say, an email address and it ended up not being used, you would end up registering an account without an email address associated, which would be surprising and potentially leave you without a means to reset your password. I therefore went with requiring all the inputs to be used.